server/server.go

```diff
@@ -743,14 +744,54 @@ func validateRodeRequirementsForPolicy(mod *ast.Module, regoContent string) []er
 	passBlockExists := false
 	// policy must contains a violations block somewhere in the code
 	violationsBlockExists := false
+	// missing field from result return response
+	returnFieldsExist := false
 
 	for _, r := range mod.Rules {
 		if r.Head.Name == "pass" {
 			passBlockExists = true
 		}
-		if r.Head.Name == "violations" && r.Head.Key != nil && r.Head.Key.Value.String() == "result" {
+
+		violations := mod.RuleSet("violations")
+		if len(violations) > 0 {
 			violationsBlockExists = true
 		}
+	violationsLoop:
+		for x, r := range violations {
+			if r.Head.Key == nil || r.Head.Key.Value.String() != "result" {
+				// found a violations block that does not return a result object, break immediately
+				break
+			}
+			for _, b := range r.Body {
+				// find the assignment
+				if b.Operator().String() == "assign" || b.Operator().String() == "eq" {
+					terms := (b.Terms).([]*ast.Term)
+					for i, t := range terms {
+						switch a := t.Value.(type) {
+						case ast.Object:
+							// look at the previous terms to check that it was assigned to result
+							if terms[i-1].String() == "result" {
+								//keys := a.Keys() // ["pass" "name" "id" "message"]
+								keys := make([]string, len(a.Keys()))
+								for i, key := range a.Keys() {
+									keys[i] = key.Value.String()
+								}
+								if !contains(keys, "\"pass\"") || !contains(keys, "\"name\"") || !contains(keys, "\"id\"") || !contains(keys, "\"message\"") {
+									break violationsLoop
```

Comment:
Anyway to restructure this to avoid the label break?

Reply:
I'll try to think of one. Usually label breaks can be avoided with some combination of breaks and continues.

Reply:
@alexashley i think the only way to avoid this is to just pull the term loop out into its own method

Reply:
Just pushed the solution. lmk what you think

Reply:
I like it!